tests/test_updater_top_level_update.py

```diff
@@ -0,0 +1,462 @@
+
+    def tearDown(self) -> None:
+        self.temp_dir.cleanup()
+
+    def _run_refresh(self) -> Updater:
+        """Create a new Updater instance and refresh"""
+        updater = Updater(
+            self.metadata_dir,
+            "https://example.com/metadata/",
+            "https://example.com/targets/",
+            fetcher=self.sim,
+        )
+        updater.refresh()
+        return updater
+
+    def _init_updater(self) -> Updater:
+        """Create a new Updater instance"""
+        return Updater(
+            self.metadata_dir,
+            "https://example.com/metadata/",
+            "https://example.com/targets/",
+            fetcher=self.sim,
+        )
+
+    def _assert_files_exist(self, roles: Iterable[str]) -> None:
+        """Assert that local metadata files exist for 'roles'"""
+        expected_files = sorted([f"{role}.json" for role in roles])
+        local_metadata_files = sorted(os.listdir(self.metadata_dir))
+        self.assertListEqual(local_metadata_files, expected_files)
+
+    def _assert_content_equals(
+        self, role: str, version: Optional[int] = None
+    ) -> None:
+        """Assert that local file content is the expected"""
+        expected_content = self.sim._fetch_metadata(role, version)
+        with open(os.path.join(self.metadata_dir, f"{role}.json"), "rb") as f:
+            self.assertEqual(f.read(), expected_content)
+
+    def test_first_time_refresh(self) -> None:
+        # Metadata dir contains only the mandatory initial root.json
+        self._assert_files_exist(["root"])
+
+        # Add one more root version to repository so that
+        # refresh() updates from local trusted root (v1) to
+        # remote root (v2)
+        self.sim.root.version += 1
+        self.sim.publish_root()
+
+        self._run_refresh()
+
+        self._assert_files_exist(TOP_LEVEL_ROLE_NAMES)
+        for role in TOP_LEVEL_ROLE_NAMES:
+            version = 2 if role == "root" else None
+            self._assert_content_equals(role, version)
+
+    def test_trusted_root_missing(self) -> None:
+        os.remove(os.path.join(self.metadata_dir, "root.json"))
+        with self.assertRaises(OSError):
+            self._run_refresh()
+
+        # Metadata dir is empty
+        self.assertFalse(os.listdir(self.metadata_dir))
+
+    def test_trusted_root_expired(self) -> None:
+        # Create an expired root version
+        self.sim.root.expires = self.past_datetime
+        self.sim.root.version += 1
+        self.sim.publish_root()
+
+        # Update to latest root which is expired but still
+        # saved as a local root.
+        updater = self._init_updater()
+        with self.assertRaises(ExpiredMetadataError):
+            updater.refresh()
+
+        self._assert_files_exist(["root"])
+        self._assert_content_equals("root", 2)
+
+        # The expiration of the trusted root metadata file does not lead
+        # to failure and Updater in successfully initialized.
```

Comment:
something wrong with sentence end here but maybe this could be shorter as well:

```suggestion
        # Local root metadata can be loaded even if expired
```

in general I really appreciate it if we manage to keep comments in tests to single lines -- makes it easier to read